cadquery/cq.py

```diff
@@ -3355,36 +3371,58 @@ def __and__(self: T, toUnion: Union["Workplane", Solid, Compound]) -> T:
 
     def cutBlind(
         self: T,
-        distanceToCut: float,
+        until: Union[float, Literal["next", "last"], Face],
         clean: bool = True,
         taper: Optional[float] = None,
     ) -> T:
         """
         Use all un-extruded wires in the parent chain to create a prismatic cut from existing solid.
+        You must define either :distance: , :untilNextFace: or :untilLastFace:
 
         Similar to extrude, except that a solid in the parent chain is required to remove material
         from. cutBlind always removes material from a part.
 
-        :param distanceToCut: distance to extrude before cutting
-        :type distanceToCut: float, >0 means in the positive direction of the workplane normal,
+        :param until: distance to extrude before cutting
+        :type until: float, >0 means in the positive direction of the workplane normal,
             <0 means in the negative direction
+        :type until: Literal, 'next' set the cut until the next face orthogonal to the wire normal, 'last' sets the extrusion until the last face
+        :type until: Face, cut until the provided face
         :param boolean clean: call :py:meth:`clean` afterwards to have a clean shape
         :param float taper: angle for optional tapered extrusion
+        :param boolean untilNextFace: cut material until next face
+        :param boolean untilLastFace: cut material last face
         :raises ValueError: if there is no solid to subtract from in the chain
         :return: a CQ object with the resulting object selected
 
         see :py:meth:`cutThruAll` to cut material from the entire part
 
-        Future Enhancements:
-            Cut Up to Surface
         """
-        # first, make the object
-        toCut = self._extrude(distanceToCut, taper=taper)
+        if isinstance(until, str):
```

Comment:
On this part : 
- I don't really understand what you want to do.
- I made some changes due to mypy that may have made it less clear than it was. I was reaffecting a variable with differents types which didn't pleased mypy.

From what you have said would it look like this ? It is more clear but not really different from the first thing

```python
        if until in ("next", "last"):
            if until == "next":
                faceIndex = 0
            elif until == "last":
                faceIndex = -1

                )
            s = self._extrude(None, taper=taper, upToFace=faceIndex, additive=False)

        elif isinstance(until, Face):
            s = self._extrude(None, taper=taper, upToFace=until, additive=False)

        elif isinstance(until, (int, float)):
            toCut = self._extrude(until, taper=taper, upToFace=None, additive=False)
            solidRef = self.findSolid()
            s = solidRef.cut(toCut)
        else:
            raise ValueError(
                "Valid option for until face extrusion are 'next' and 'last'"

        if clean:
            s = s.clean()

        return self.newObject([s])
```
- As I have a preference for `next` and you seems to have a preference for `first` it could be interesting to have the opinion of @jmwright  and @marcus7070 on the topic

Reply:
`next` is fine, the comment was regarding code structuring

Reply:
Do you know a use case when extrude(somevalue) with some value being a int or a float would raise an error ? If yes then I'll add it in the `raise` string to give more feedback to the user

Reply:
Still not clear - if `_extrude` raises the user will see the error. No need to catch and rethrow (assuming that this is what you meant)